File: GC.Domain/Contacts/AllContacts.cs
using GC.Domain.Contacts.EmergencyContacts;
using GC.Domain.Contacts.ForeignContacts;
using GC.Domain.Contacts.GardenContacts;

namespace GC.Domain.Contacts
{
    public class AllContacts
    {
        public EmergencyContact[] EmergencyContacts { get; }
        public ForeignContact[] ForeignContacts { get; }
        public GardenContact[] GardenContacts { get; }

        public AllContacts(EmergencyContact[] emergencyContacts, ForeignContact[] foreignContacts, GardenContact[] gardenContacts)
        {
            EmergencyContacts = emergencyContacts;
            ForeignContacts = foreignContacts;
            GardenContacts = gardenContacts;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/contacts/allContacts.ts
import { EmergencyContact } from "./emergencyContacts/emergencyContact";
import { ForeignContact } from "./foreignContacts/foreignContact";
import { GardenContact } from "./gardenContacts/gardenContact";

export class AllContacts {
    constructor(
        public readonly emergencyContacts: EmergencyContact[],
        public readonly foreignContacts: ForeignContact[],
        public readonly gardenContacts: GardenContact[],
    ) { }
}

export namespace AllContacts {
    export function toModel(object: any): AllContacts {
        return new AllContacts(
            EmergencyContact.toModels(object.emergencyContacts as any[]),
            ForeignContact.toModels(object.foreignContacts as any[]),
            GardenContact.toModels(object.gardenContacts as any[]));
    }
}